Update how-to-map-hresults-and-exceptions.md

jkotas opened the pull request:
Fixes #18616

## Summary

- Clarify that the mapping is not complete
- Add link to the current complete mapping in the runtime sources
- Delete HRESULT mappings that either do not exist or that are corner cases

Fixes #18616
Fixes #18615

jkotas committed:
Update how-to-map-hresults-and-exceptions.md

Fixes #18616

jkotas commented on docs/framework/interop/how-to-map-hresults-and-exceptions.md:
````diff
@@ -49,36 +49,29 @@ CMyClass::MethodThatThrows
 ```  
   
- The following table provides the complete mapping from each HRESULT to its comparable exception class in the .NET Framework.  
+ The following table provides the common mappings from HRESULT to its comparable exception class in the .NET.  HRESULT values without explicit mappings are mapped to **COMException**.  The complete up-to-date mapping can be found in [dotnet/runtime repo](https://github.com/dotnet/runtime/blob/master/src/coreclr/vm/rexcep.h). 
````
```suggestion
 The following table provides the common mappings from HRESULT to its comparable exception class in the .NET. HRESULT values without explicit mappings are mapped to **COMException**.  The complete up-to-date mapping can be found in [dotnet/runtime repo](https://github.com/dotnet/runtime/blob/master/src/coreclr/vm/rexcep.h). 
```

jkotas committed:
Update docs/framework/interop/how-to-map-hresults-and-exceptions.md

jkotas commented on docs/framework/interop/how-to-map-hresults-and-exceptions.md:
````diff
@@ -49,36 +49,29 @@ CMyClass::MethodThatThrows
 ```  
   
- The following table provides the complete mapping from each HRESULT to its comparable exception class in the .NET Framework.  
+ The following table provides the common mappings from HRESULT to its comparable exception class in the .NET. HRESULT values without explicit mappings are mapped to **COMException**.  The complete up-to-date mapping can be found in [dotnet/runtime repo](https://github.com/dotnet/runtime/blob/master/src/coreclr/vm/rexcep.h). 
````
```suggestion
 The following table provides the common mappings from HRESULT to its comparable exception class in the .NET. HRESULT values without explicit mappings are mapped to **COMException**. The complete up-to-date mapping can be found in [dotnet/runtime repo](https://github.com/dotnet/runtime/blob/master/src/coreclr/vm/rexcep.h). 
```

jkotas committed:
Update docs/framework/interop/how-to-map-hresults-and-exceptions.md

jkotas commented on docs/framework/interop/how-to-map-hresults-and-exceptions.md:
````diff
@@ -49,36 +49,29 @@ CMyClass::MethodThatThrows
 ```  
   
- The following table provides the complete mapping from each HRESULT to its comparable exception class in the .NET Framework.  
+ The following table provides the common mappings from HRESULT to its comparable exception class in the .NET. HRESULT values without explicit mappings are mapped to **COMException**. The complete up-to-date mapping can be found in [dotnet/runtime repo](https://github.com/dotnet/runtime/blob/master/src/coreclr/vm/rexcep.h). 
````
I am not able to figure out what the markdown lint is complaining about.

adegeo replied:
It's mad that you have a space at the end of the line, you should have 0. Most articles that were automatically ported added 2 spaces to all lines, which generally get cleaned up when the article gets a full refresh. But 0 is what you should use.

jkotas replied:
> It's mad that you have a space at the end of the line, you should have 0. 

I have removed the trailing whitespaces from the whole file. Use https://github.com/dotnet/docs/pull/22487/files?w=1 to see the real delta without the whipespace only changes.

adegeo reviewed (commented):
I notice that you changed some wording from **.NET Framework** to **.NET**, yet this article was written for framework. Should there also be an article similar to this but for .NET? Perhaps in this area: https://docs.microsoft.com/en-us/dotnet/standard/native-interop/exceptions-interoperability

jkotas committed:
Update docs/framework/interop/how-to-map-hresults-and-exceptions.md

Co-authored-by: Tom Dykstra <tdykstra@microsoft.com>

jkotas committed:
Consistent formatting

- Remove trailing whitespaces
- Use backticks for type names and code fragments

jkotas commented:
This article is equally applicable for .NET Framework and .NET Core. Should it be moved somewhere?

tdykstra commented on docs/framework/interop/how-to-map-hresults-and-exceptions.md:
```diff
@@ -15,127 +15,112 @@ ms.assetid: 610b364b-2761-429d-9c4a-afbc3e66f1b9
+|HRESULT|.NET exception|
+|-------------|--------------------|
+|`COR_E_APPLICATION`|`ApplicationException`|
+|`COR_E_ARGUMENT or E_INVALIDARG`|`ArgumentException`|
```
The "or" shouldn't be included in the code format.  That would prevent it from being localized.
```suggestion
|`COR_E_ARGUMENT` or `E_INVALIDARG`|`ArgumentException`|
```

jkotas committed:
Update docs/framework/interop/how-to-map-hresults-and-exceptions.md

Co-authored-by: Tom Dykstra <tdykstra@microsoft.com>

tdykstra commented:
>This article is equally applicable for .NET Framework and .NET Core. Should it be moved somewhere?

I believe it would fit better in the [.NET Fundamentals/Native interop TOC](https://github.com/dotnet/docs/blob/868ca7cd5802603d19f02e6e7994511eb8ad873f/docs/fundamentals/toc.yml#L2118).

jkotas commented:
I would prefer it to be done in separate change. There are multiple articles under framework/interop that are applicable to .NET Core. There may be also a good reason to keep them where they are. While these articles are applicable to .NET Core, they talk about legacy features that were brought over to .NET Core just for compatibility.

tdykstra commented:
Sounds good. Is this ready to merge, then?

jkotas commented:
I think so.

tdykstra reviewed (approved):
LGTM
